torchaudio/functional/functional.py

```diff
@@ -1631,3 +1637,274 @@ def rnnt_loss(
+
+    Args:
+        rtf (Tensor): RTF vector of target speech.
+            Tensor of dimension `(..., freq, channel)`.
+        psd_n (torch.Tensor): The covariance matrix of noise.
+            Tensor of dimension `(..., freq, channel, channel)`
+        reference_channel (int or Tensor): Indicate the reference channel.
+            If the dtype is ``int``, it represent the reference channel index.
+            If the dtype is ``Tensor``, the dimension is `(..., channel)`.
+        normalize (bool, optional): whether to normalize the RTF vector. (Default: ``True``)
+        diagonal_loading (bool, optional): whether to apply diagonal loading to psd_n
+            (Default: ``True``)
+        diag_eps (float, optional): The coefficient multipied to the identity matrix for diagonal loading
+            (Default: ``1e-7``)
+        eps (float, optional): a value added to the denominator in mask normalization. (Default: ``1e-8``)
+
+    Returns:
+        Tensor: The MVDR beamforming weight matrix of dimension (..., freq, channel).
+    """
+    if diagonal_loading:
+        psd_n = _tik_reg(psd_n, reg=diag_eps, eps=eps)
+    if isinstance(reference_channel, int):
+        ref_vector = torch.zeros(
+            psd_n.size()[:-3] + psd_n.size()[-1:], device=psd_n.device, dtype=psd_n.dtype
+        )  # (..., channel)
+        ref_vector[..., reference_channel].fill_(1)
+    else:
+        ref_vector = reference_channel
+    # numerator = psd_n.inv() @ stv
+    numerator = torch.linalg.solve(psd_n, rtf).squeeze(-1)  # (..., freq, channel)
+    # denominator = stv^H @ psd_n.inv() @ stv
+    denominator = torch.einsum("...d,...d->...", [rtf.conj().squeeze(-1), numerator])
+    beamform_weights = numerator / (denominator.real.unsqueeze(-1) + eps)
+    # normalzie the numerator
+    if normalize:
+        scale = torch.einsum("...c,...c->...", [rtf.conj().squeeze(-1), ref_vector[..., None, :]])
+        beamform_weights = beamform_weights * scale[..., None]
+
+    return beamform_weights
+
+
+def compute_rtf_evd(psd_s: Tensor) -> Tensor:
+    r"""Estimate the relative transfer function (RTF) or the steering vector by eigenvalue decomposition.
+
+    Args:
+        psd_s (Tensor): The covariance matrix of target speech.
+            Tensor of dimension `(..., freq, channel, channel)`
+
+    Returns:
+        Tensor: The estimated RTF of target speech.
+            Tensor of dimension `(..., freq, channel, 1)`
+    """
+    w, v = torch.linalg.eig(psd_s)  # (..., freq, channel, channel)
+    _, indices = torch.max(w.abs(), dim=-1, keepdim=True)
+    indices = indices.unsqueeze(-1)
+    rtf = v.gather(-1, indices.expand(psd_s.shape[:-1] + (1,)))  # (..., freq, channel, 1)
+    return rtf
+
+
+def compute_rtf_power(psd_s: Tensor, psd_n: Tensor, reference_channel: Union[int, Tensor]) -> Tensor:
+    r"""Estimate the relative transfer function (RTF) or the steering vector by the power method.
```

Comment:
I miss an iteration argument or is power method not the same as power iteration?